mono/arch/x86/x86-codegen.h

```diff
@@ -309,7 +316,7 @@ typedef union {
 		x86_address_byte ((inst), 0, (r), (basereg));	\
```

Comment:
Might need to disallow the 0 also, under single_imm_size.

Reply:
That can be done separately if needed.
I'd rather keep this PR reduced to "imm8" and ignore "0".